src/service/poc_mobile.proto

```diff
@@ -106,22 +86,31 @@ enum share_validity {
   share_validity_not_operational = 5;
 }
 
-message moving_avg {
+message speedtest_avg {
   bytes pub_key = 1;
-  /// Refers to the number of speed_shares used to calculate the average
-  uint64 window_size = 2;
-  average average = 3;
-  bool is_valid = 4;
-  repeated speed_share speed_shares = 5;
+  /// bytes/sec
+  uint64 upload_speed_avg_bps = 2;
+  /// bytes/sec
+  uint64 download_speed_avg_bps = 3;
+  uint32 latency_avg_ms = 4;
+  speedtest_avg_validity validity = 5;
+  repeated speedtest speedtests = 6;
 }
 
-message average {
-  uint64 upload_speed_avg_bps = 1;
-  uint64 download_speed_avg_bps = 2;
-  uint32 latency_avg_ms = 3;
+message speedtest {
+  uint64 upload_speed_bps = 1;
+  uint64 download_speed_bps = 2;
+  uint32 latency_ms = 3;
+  uint64 timestamp = 4;
 }
 
-message speed_share_moving_avgs { repeated moving_avg moving_avgs = 1; }
+enum speedtest_avg_validity {
+  speedtest_avg_validity_valid = 0;
```

Comment:
Is this value derived from either of these being true speedtest_avg_validity_slow_download_speed, speedtest_avg_validity_slow_upload_speed, speedtest_avg_validity_high_latency?

Reply:
That value would be if none of the others are true